components/accordion/index.jsx

```diff
@@ -45,17 +45,55 @@ const propTypes = {
  * The accordion component wraps accordion panels that can be selected and expanded. It accepts children to define the content displayed within.
  */
 class Accordion extends Component {
+	constructor(props) {
+		super(props);
+		this.state = { currButtonIndex: 0 };
+	}
+
 	componentWillMount() {
 		this.generatedId = shortid.generate();
 	}
 
+	componentDidUpdate(prevState) {
+		if (this.state.currButtonIndex !== prevState.currButtonIndex) {
+			const { id } = this.props.children[this.state.currButtonIndex].props;
+			document.getElementById(`${id}-accordion-button`).focus();
```

Comment:
Please do not use DOM query selection here. Please use React refs. An alternative approach would be by passing a function down into the Panel (`refs.summaryButton`) and the the panel passing a ref from the render function back up into the parent. I think the refs would be stored in an array. Here is radio's input ref for example: https://github.com/salesforce/design-system-react/blob/master/components/radio/index.jsx#L107

Two other more complicated options are:
* The other option would be to render each panel with a `hasFocus=true` prop and that typically involves a second flag that says whether any of the buttons are currently focused. The button would only focus if both flags were true, so that we only get one `focus` call. Datepicker does this type of approach: https://github.com/salesforce/design-system-react/blob/master/components/date-picker/private/calendar.jsx#L82
* The way that the Combobox menu items work is with an `active == props.activeOptionIndex` compare. You'd pass the active panel down.